src/Logging/Logging.Testing/src/CollectDumpAttribute.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Logging.Testing
+{
+    /// <summary>
+    /// Capture the memory dump upon test failure.
+    /// </summary>
+    /// <remarks>
+    /// This currently only works in Windows environments
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
```

Comment:
Currently setting this to method only since only few tests should need this. Could expandable to class/assembly if we want.